etl/steps/data/garden/who/2024-04-08/polio.py

```diff
@@ -0,0 +1,326 @@
+    If both 'non_polio_afp_rate' and 'percent_adequate_stool_collection' are missing then the correction factor is 7.
+
+    There are some manual changes we make:
+
+    - Namibia had 'percent_adequate_stool_collection' > 100 in 2011 and 2014 but for other years it's correction factor is 1.11 so we set it as 1.11 for 2011 and 2014.
+
+    - For China 1989-92 we set the correction factor to 1.11 and in Oman in 1988.
+
+    (We set the correction factor as NA for all of 2021 as the values of 'percent_adequate_stool_collection' seemed unreliable in this year.)
+
+    """
+    # tb["correction_factor"] = pd.NA
+    # Correction factor for years 1996-2000 is 1.11.
+    tb.loc[(tb["year"] >= 1996) & (tb["year"] <= 2000), "correction_factor"] = 1.11
+    # If the 'non_polio_afp_rate' is < 1 OR 'percent_adequate_stool_collection' < 60, then the correction factor = 7.
+    tb.loc[(tb["non_polio_afp_rate"] < 1.0) | (tb["pct_adequate_stool_collection"] < 60), "correction_factor"] = 7.0
+    # If the 'non_polio_afp_rate' is < 2 OR 'percent_adequate_stool_collection' < 80, then the correction factor = 2.
+    tb.loc[(tb["non_polio_afp_rate"] < 2.0) | (tb["pct_adequate_stool_collection"] < 80), "correction_factor"] = 2.0
+    # If the 'non_polio_afp_rate' is >= 2 OR 'percent_adequate_stool_collection' >= 80, then the correction factor = 1.11.
+    tb.loc[(tb["non_polio_afp_rate"] >= 2.0) & (tb["pct_adequate_stool_collection"] >= 80), "correction_factor"] = 1.11
+    # If both 'non_polio_afp_rate' and 'percent_adequate_stool_collection' are missing then the correction factor is 7.
+    tb.loc[(tb["non_polio_afp_rate"].isna()) & (tb["pct_adequate_stool_collection"].isna()), "correction_factor"] = 7.0
+    # Correction factor for years before 1996 is 7.
+    tb.loc[tb["year"] < 1996, "correction_factor"] = 7.0
+
+    # tb.loc[tb["year"] == 2021, "correction_factor"] = np.nan
+
+    # Namibia had 'percent_adequate_stool_collection' > 100 in 2011 and 2014 but for other years it's correction factor is 1.11 so we set it as 1.11 for 2011 and 2014.
+    tb.loc[(tb["country"] == "Namibia") & (tb["year"].isin([2011, 2014])), "correction_factor"] = 1.11
+    # For China 1989-92 we set the correction factor to 1.11 and in Oman in 1988.
+    tb.loc[(tb["country"] == "China") & (tb["year"].isin([1989, 1990, 1991, 1992])), "correction_factor"] = 1.11
+    tb.loc[(tb["country"] == "Oman") & (tb["year"].isin([1988])), "correction_factor"] = 1.11
+    # Not sure if this is the best way to handle this, the code fails because this indicator doesn't have origins otherwise
+    tb["correction_factor"].metadata.origins = tb["non_polio_afp_rate"].metadata.origins
+    return tb
+
+
+def clean_adequate_stool_collection(tb: Table) -> Table:
+    """
+    Some values for "Adequate stool collection" are over 100%, we should set these to NA.
+    """
+    tb.loc[tb["pct_adequate_stool_collection"] > 100, "pct_adequate_stool_collection"] = pd.NA
```

Comment:
It seems to happen only in three cases, with values 102, 104 and 113. I haven't looked into it, but I'm wondering if it would be better to set 102 and 104 to 100, and consider it a numerical issue with rounded numbers (and 113 to nan, as it possibly is an issue). Up to you.

Reply:
Will check with Saloni!